templates/macros/https_params.conf

```diff
@@ -0,0 +1,23 @@
+{% macro params() %}
+################################################################################################################
+# https parameters
+# https://ssl-config.mozilla.org/#server=nginx&version=1.18.0&config=intermediate&openssl=1.1.1f&guideline=5.6 #
+################################################################################################################
+
+ssl_session_timeout 1d;
+ssl_session_cache shared:MozSSL:50m;
+ssl_session_tickets off;
+
+ssl_protocols TLSv1.2 TLSv1.3;
+ssl_ciphers ECDHE-ECDSA-AES128-GCM-SHA256:ECDHE-RSA-AES128-GCM-SHA256:ECDHE-ECDSA-AES256-GCM-SHA384:ECDHE-RSA-AES256-GCM-SHA384:ECDHE-ECDSA-CHACHA20-POLY1305:ECDHE-RSA-CHACHA20-POLY1305:DHE-RSA-AES128-GCM-SHA256:DHE-RSA-AES256-GCM-SHA384;
+ssl_prefer_server_ciphers off;
+
+ssl_dhparam {{ nginx.prefix.config }}/dhparam.pem;
+
+ssl_stapling on;
+ssl_stapling_verify on;
+
+{% if nginx.hsts %}
+add_header Strict-Transport-Security "max-age={{ nginx.hsts.max_age }}{% if nginx.hsts.include_subdomains %}; includeSubDomains{% endif %}{% if nginx.hsts.preload %}; preload{% endif %}";
```

Comment:
Since this is now a macro, it should be possible to override the defaults using parameters.
```j2
{% macro https_params(hsts=nginx.hsts) %}
```
I don't think we need to add a parameter for everything but since `nginx.hsts` is already a config option... It would be nice if we could enable/disable HSTS per vhost.

Reply:
I had this at first but, changed it back due to compatibility reasons, bc that would mean we had to change the variables